src/tile.rs

```diff
@@ -2,26 +2,187 @@ use bytes::Bytes;
+            // last chunk
+            Ok(self.image_width - self.chunk_width * x)
+        } else {
+            Ok(self.chunk_width)
+        }
+    }
+
+    /// chunk height in pixels, taking padding into account
+    ///
+    /// strips are considered image-width chunks
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use async_tiff::tiff::tags::{SampleFormat, PlanarConfiguration};
+    /// # use async_tiff::reader::Endianness;
+    /// # use async_tiff::PredictorInfo;
+    /// let info = PredictorInfo {
+    /// # endianness: Endianness::LittleEndian,
+    ///   image_width: 15,
+    ///   image_height: 15,
+    ///   chunk_width: 8,
+    ///   chunk_height: 8,
+    /// # bits_per_sample: &[32],
+    /// # samples_per_pixel: 1,
+    /// # sample_format: &[SampleFormat::IEEEFP],
+    /// # planar_configuration: PlanarConfiguration::Chunky,
+    /// };
+    ///
+    /// assert_eq!(info.chunk_height_pixels(0).unwrap(), (8));
+    /// assert_eq!(info.chunk_height_pixels(1).unwrap(), (7));
+    /// info.chunk_height_pixels(2).unwrap_err();
+    /// ```
+    pub fn chunk_height_pixels(&self, y: u32) -> AsyncTiffResult<u32> {
+        if y >= self.chunks_down() {
+            Err(crate::error::AsyncTiffError::TileIndexError(
+                y,
+                self.chunks_down(),
+            ))
+        } else if y == self.chunks_down() - 1 {
+            // last chunk
+            Ok(self.image_height - self.chunk_height * y)
+        } else {
+            Ok(self.chunk_height)
+        }
+    }
+
+    /// get the output row stride in bytes, taking padding into account
+    pub fn output_row_stride(&self, x: u32) -> AsyncTiffResult<usize> {
+        Ok((self.chunk_width_pixels(x)? as usize).saturating_mul(self.bits_per_pixel()) / 8)
+    }
+
+    /// The total number of bits per pixel, taking into account possible different sample sizes
+    ///
+    /// Technically bits_per_sample.len() should be *equal* to samples, but libtiff also allows
+    /// it to be a single value that applies to all samples.
+    ///
+    /// Libtiff and image-tiff do not support mixed bits per sample, but we give the possibility
+    /// unless you also have PlanarConfiguration::Planar, at which point the first is taken
+    pub fn bits_per_pixel(&self) -> usize {
+        match self.planar_configuration {
+            PlanarConfiguration::Chunky => {
+                if self.bits_per_sample.len() == 1 {
+                    self.samples_per_pixel as usize * self.bits_per_sample[0] as usize
+                } else {
+                    assert_eq!(self.samples_per_pixel as usize, self.bits_per_sample.len());
+                    self.bits_per_sample.iter().map(|v| *v as usize).product()
+                }
+            }
+            PlanarConfiguration::Planar => self.bits_per_sample[0] as usize,
+        }
+    }
+
+    /// The number of chunks in the horizontal (x) direction
+    pub fn chunks_across(&self) -> u32 {
```

Comment:
To add to my previous comment on removing the `PredictorInfo` struct; both `chunks_across` and `chunks_down` are useful outside of handling predictors.  It would make sense to include these on the `IFD` struct (ex. `IFD.chunks_across`).

Reply:
Even if we keep the `PredictorInfo`, as I think we probably should, we can put this functionality in a shared `TileMath` trait which both the IFD and `PredictorInfo` implement